cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/templates/etl/realtime/sources/JmsSource.java

```diff
@@ -95,13 +99,17 @@ public JmsSource(JmsPluginConfig config) {
    */
   public void initialize(RealtimeContext context) throws Exception {
     super.initialize(context);
-    Map<String, String> runtimeArguments = context.getPluginProperties().getProperties();
+    Map<String, String> runtimeArguments = Maps.newHashMap();
+    if (config.getProperties() != null) {
+      runtimeArguments.putAll(config.getProperties().getProperties());
+    }
 
     Integer configMessagesToReceive = config.messagesToReceive;
     messagesToReceive = configMessagesToReceive.intValue();
 
-    // Try get the destination name
-    String destinationName = config.destinationName;
+    // Set initial context factory name and provider URL
+    runtimeArguments.put(Context.INITIAL_CONTEXT_FACTORY, config.initialContextFactory);
+    runtimeArguments.put(Context.PROVIDER_URL, config.providerUrl);
 
     // Get environment vars - this would be prefixed with java.naming.*
     final Hashtable<String, String> envVars = new Hashtable<String, String>();
```

Comment:
This will eventually be passed to JmsProvider which later be pushed to InitialContext which only take HashTable. Either do HashTable now or convert later =(

This  is what happen when we use 10 year old tech =P